lib/mix/tasks/nerves_hub.key.ex

```diff
@@ -5,41 +5,68 @@ defmodule Mix.Tasks.NervesHub.Key do
   alias NervesHubCLI.API
   alias Mix.NervesHubCLI.Shell
 
-  @shortdoc "Manages your NervesHub keys"
+  @shortdoc "Manages your firmware signing keys"
 
   @moduledoc """
-  Manage your NervesHub keys.
+  Manages your firmware signing keys.
+
+  Firmware signing keys consist of public and private keys. The `mix
+  nerves_hub.key` task manages both pieces for you. Private signing keys are
+  password-protected and are NEVER sent to NervesHub or any other server.
+  Public keys, however, are registered with NervesHub.
+
+  Signing keys are stored in `~/.nerves-hub/keys`. Keys may be shared between
+  developers by copying the files in this folder.
+
+  NervesHub can manage more than one key so that you can have different
+  development and production keys in use. For example, production devices
+  deployed with only the production public key will not accept firmware signed
+  by development keys.
+
+  To ensure that firmware includes keys registered with NervesHub, add the
+  following entries in your project's `config.exs`:
+
+    # Specify that NervesHub hub provides device provisioning metadata
+    config :nerves, :firmware,
+      provisioning: :nerves_hub
```

Comment:
Adding provisioning is technically only required to get the overall connection certificates onto the device. I am unsure if it belongs in this part of the docs, but it should be stressed somewhere.